### DISCUSSION TITLE
Where are signed 8-bit and 16-bit numbers?

### DISCUSSION THREAD

**Discussion by hirrolot** [yallop/ocaml-integers#46] [Q&A] [answered]
In the `Signed` module, I see only the `Int32` and `Int64` types, but there is no `Int8` and `Int16`. Are they simply not implemented or I search in a wrong place? From `README.md`, it follows that `Int8` and `Int16` should be implemented:

> The ocaml-integers library provides a number of 8-, 16-, 32- and 64-bit signed and unsigned integer types

**Comment by yallop** [comment]
They're currently unimplemented.  A PR would be welcome.

**Comment by brendanzab** [comment]
This would be pretty helpful for making programming languages!

Edit: Seems like there is a PR about it here: https://github.com/yallop/ocaml-integers/pull/49?

**Comment by brendanzab** [comment]
[int_repr](https://ocaml.org/p/int_repr) seems like an alternative package that supports the full complement of integers up to 128 bits. Unfortunately it requires you to depend on Jane Street’s standard library, as [mentioned in this thread](https://discuss.ocaml.org/t/idiomatic-ocaml-int8-int16-int32-int64-uint8-uint16-uint32-uint64/11754/7).

**Reply by zoj613** [reply]
@brendanzab the documentation of int_repr seems to suggest it supports ints up to 64 bit. Could you please link me to the module implementing 128 bit width?

**Comment by yallop** [comment, marked as answer]
#54 (now merged) adds support for 8- and 16-bit signed integers
